master/internal/experiment.go

```diff
@@ -418,13 +418,12 @@ func (e *internalExperiment) SetGroupMaxSlots(msg sproto.SetGroupMaxSlots) {
 		return
 	}
 
-	slots, err := configpolicy.CanSetMaxSlots(msg.MaxSlots, w.ID)
+	err = configpolicy.CanSetMaxSlots(msg.MaxSlots, w.ID)
```

Comment:
This PR changes/undoes some of the changes you just made in your previous PR ... why is that?

Reply:
This was my bad, when I made this PR I didn't realize that a lot of these PATCH changes were created w the `det e set` CLI commands. After testing PATCH requests resulting from those commands and adding automated testing for such requests, I think these changes _should_ properly address any remaining issues/inconsistencies with expected config policy-related behavior

Reply:
This was my bad, when I made the previous PR I didn't realize that a lot of the code in the PATCH experiment API handler is specifically for `det e set` CLI commands. After testing PATCH requests resulting from those commands and adding automated integration testing for such requests, I think these changes _should_ properly address any remaining issues/inconsistencies with expected config policy-related behavior in this API handler

Reply:
i'm still slightly fuzzy here. just to make sure we're all on the same page.. `resources.max_slots/slots/slots_per_trial` are all different but we're ok with `constraints.max_slots` controlling all of them? in which case.. maybe if there is a constraint or invariant for max slots it should always get set here, because otherwise an experiment could exceed max slots?

Reply:
Yea! This is just for patching the `resources.max_slots` field specifically. Somewhere in the request validation, it could be useful to check `resouces.slots` or `resouces.slots_per_trial` in this case to make sure it doesn't exceed the requested `resources.max_slots`, but I think this goes back to the conversation of not wanting a constraint to alter configs? I think it could be worth checking in this API handler that `resources.max_slots` >= `resources.slots_per_trial` in the experiment config, but that kinda goes beyond intended fixes for this ticket
Happy to add it here either way though, since a check like that does make sense!